lib/core/src/Data/Quantity.hs

```diff
@@ -134,49 +136,82 @@ instance (KnownSymbol unit, Buildable a) => Buildable (Quantity unit a) where
 newtype Percentage = Percentage
-    { getPercentage :: Word }
+    { getPercentage :: Rational }
     deriving stock (Generic, Show, Eq, Ord)
-    deriving newtype (ToJSON)
+
+-- | Desired number of digits after the decimal point for presenting the
+-- @Percentage@ type.
+percentageNumberOfFractionalDigits :: Int
+percentageNumberOfFractionalDigits = 2
+
+-- | Round a @Rational@ to the given amount of fractional digits.
+--
+-- Note: This is safe to call on repeating digits, in contrast to @fromRational@
+-- (for creating a @Scientific@).
+roundToScientific :: Int -> Rational -> Scientific
+roundToScientific fracDigits x = (conv i) / (conv factor)
+  where
+    i :: Int
+    i = round (factor * (toRational x))
+
+    conv :: (Real a, Fractional b) => a -> b
+    conv = fromRational . toRational
+
+    factor = 10 ^ fracDigits
 
 instance NFData Percentage
 
+instance ToJSON Percentage where
+    toJSON =
+        toJSON
+        . roundToScientific percentageNumberOfFractionalDigits
+        . getPercentage
+
 instance FromJSON Percentage where
-    parseJSON x = do
-        n <- parseJSON x
-        either (fail . show) return (mkPercentage @Int n)
+    parseJSON = withScientific "Percentage [0,100]" $ \s ->
+        either (fail . show) return $ mkPercentage (toRational s/100)
 
 instance Bounded Percentage where
     minBound = Percentage 0
     maxBound = Percentage 100
 
-instance Enum Percentage where
-    fromEnum (Percentage p) = fromEnum p
-    toEnum = either (error . ("toEnum: " <>) . show) id . mkPercentage
-
 instance ToText Percentage where
-    toText (Percentage p) = T.pack (show p) <> "%"
+    toText (Percentage p) =
+        (<> "%")
+        . T.pack
+        . showS
+        . roundToScientific percentageNumberOfFractionalDigits
+        $ p
+      where
+        showS = formatScientific
+            Fixed
+            (Just percentageNumberOfFractionalDigits)
 
 instance FromText Percentage where
     fromText txt = do
-        (p, u) <- left (const err) $ decimal txt
+        (p, u) <- left (const err) $ rational txt
         unless (u == "%") $ Left err
-        left (const err) $ mkPercentage @Integer p
+        left (const err) . mkPercentage $ (p / 100)
       where
         err = TextDecodingError
             "expected a value between 0 and 100 with a '%' suffix (e.g. '14%')"
 
 -- | Safe constructor for 'Percentage'
+--
+-- Takes an input in the range [0, 1].
 mkPercentage
-    :: Integral i
-    => i
+    :: Rational
     -> Either MkPercentageError Percentage
-mkPercentage i
-    | let Percentage inf = minBound in i < fromIntegral inf =
+mkPercentage r
+    | r < 0 =
         Left PercentageOutOfBoundsError
-    | let Percentage sup = maxBound in i > fromIntegral sup =
+    | r > 1 =
         Left PercentageOutOfBoundsError
     | otherwise =
-        pure $ Percentage $ fromIntegral i
+        let
+            toPercent = (* 100)
+        in
+            pure . Percentage . toPercent $ r
```

Comment:
Why the multiplication here?
Representing percentage as values between 0 and 100 is formatting concern, so it should only happens within the `toText` or `toJSON` instance, it's quite error prone to store a 100x the actual value :/